### PR TITLE
Add dockcross for C++ x86 cross compilation.

### PR DESCRIPTION
**Pull request by Enkelmann**
We could also use dockcross containers for all other cross compiling targets if we want to.

### CODE REVIEW

File: test/artificial_samples/install_cross_compilers_ubuntu_18.04.sh [hunk 1]
```diff
@@ -5,4 +5,6 @@ echo "Installing cross compiler for MIPS architecture."
 sudo apt install -y gcc-multilib-mips-linux-gnu g++-7-mips-linux-gnu
```
Comment (tbarabosch) [inline test/artificial_samples/install_cross_compilers_ubuntu_18.04.sh]: can't we cross compile everything with dockcross? By doing so, we would not need to pollute our systems with the cross compiling toolchains of arm, mips, and so on.

Comment (tbarabosch) [reply test/artificial_samples/install_cross_compilers_ubuntu_18.04.sh]: if this is correct, then we can remove the other dependencies.

File: test/artificial_samples/makefile [hunk 2]
```diff
@@ -36,7 +36,7 @@ endef
```
Comment (tbarabosch) [inline test/artificial_samples/makefile]: can't we change everything to dockcross?

File: test/artificial_samples/dockcross_x86 [hunk 3]
```diff
@@ -0,0 +1,254 @@
+#!/usr/bin/env bash
```
[the remark this answers is not in the document]
Comment (Enkelmann) [reply test/artificial_samples/dockcross_x86]: was auto-generated from dockcross, now the install script generates it.

**Comment by Enkelmann** [comment]
I have replaced all cross compiling toolchains with dockcross. The downside is the higher disk space required (about 4GB alltogether, can probably be optimized if we care about it).

File: test/artificial_samples/makefile (hunk 2 above)
Comment (Enkelmann) [reply test/artificial_samples/makefile]: Done.

File: test/artificial_samples/install_cross_compilers_ubuntu_18.04.sh (hunk 1 above)
Comment (Enkelmann) [reply test/artificial_samples/install_cross_compilers_ubuntu_18.04.sh]: Done.

**Pull request merged**